processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java

```diff
@@ -197,11 +201,13 @@ else if ( matchingParameterAssignmentVariants.size() == 1 ) {
             // we had some matching assignments before, but when checking the parameter names we can't find an
             // appropriate one, in this case the user must chose identical parameter names for the mapping and lifecycle
             // method
-            messager.printMessage(
-                selectedMethod.getExecutable(),
-                Message.LIFECYCLEMETHOD_AMBIGUOUS_PARAMETERS,
-                mappingMethod
-            );
+            if ( !warningAlreadyGiven( mappingMethod, selectedMethod ) ) {
```

Comment:
under which condition does this occur?  When I read the comments my guess is it has some conflicting method signatures with the same parameters and it cannot select the right one based on names.

So for instance:

`A methodX(B one, C two);`
`A methodX(C three, B four)`

And the calling method does not have similar parameter names.


Could it be that the error is than given for both methods? I guess that's correct. Right?

Reply:
it runs twice through the code, which is why the warning was suddenly given twice. fixing the double run through the code would also reduce this warning to only once.

Reply:
As to when the warning occurs:

When you have as mapping method: 
`A methodX(String something, String somethingElse)`
and as matching method: 
`A methodY(String totallyDifferent, String alsoTotallyDifferent)`
As mapstruct cannot decide which 2 variables should match.

With mapping method:
`A methodX(B one, C two)`
and matching method:
`A methodX(B three, C four)`
this does not occur since the types are already unique enough to identify how to match.